sregistry/main/registry/auth.py

```diff
@@ -16,25 +16,44 @@
     generate_timestamp
 )
 
+from sregistry.logger import bot
+
+def _check_secret(cli, secret):
+    '''sets a secret attribute on the registry client, based
+    on the value of a corresponding environment variable'''
+
+    env_secret = 'SREGISTRY_REGISTRY_{}'.format(secret.upper())
+
+    setattr(cli, secret, cli._get_and_update_setting(env_secret))
+
+    if getattr(cli, secret, None) is None:
+        # backwards compatibility
+        setattr(cli, secret, cli._get_and_update_setting(secret))
+
+        if getattr(cli, secret, None) is None:
+            bot.exit('Failed to authorize: please set {} to an appropriate value'.format(env_secret))
+        else:
+            cli._update_setting(env_secret, getattr(cli, secret))
+
+
```

Comment:
Sorry this is likely frustrating - I think we should restore the simpler version with redundancy. Adding this extra layer is mostly confusing for some future developer reading it. I favor the redundancy over that.

Reply:
Ok no worries